test/simplify_qdq_test.cpp

```diff
@@ -0,0 +1,748 @@
+        auto weights = m2->add_parameter("weights", sw);
+        auto scale   = m2->add_literal(0.1f);
+        auto zero    = m2->add_literal(std::int8_t{0});
+
+        auto d1 = add_quantize_op(*m2, "dequantizelinear", weights, scale, zero);
+        auto c1 = m2->add_instruction(migraphx::make_op("convolution",
+                                                        {{"padding", {0, 0, 0, 0}},
+                                                         {"stride", {1, 1}},
+                                                         {"dilation", {1, 1}},
+                                                         {"group", 1},
+                                                         {"padding_mode", 0}}),
+                                      input,
+                                      d1);
+        m2->add_return({c1});
+    }
+
+    std::vector<float> iv(si.elements(), 4);
+    auto input = migraphx::argument(si, iv.data());
+    std::vector<float> wv(sw.elements(), 10);
+    auto weights = migraphx::argument(sw, wv.data());
+    p1.compile(migraphx::target(migraphx::ref::target{}));
+    p2.compile(migraphx::target(migraphx::ref::target{}));
+
+    auto result1 = p1.eval({{"input", input}, {"weights", weights}}).back();
+    std::vector<float> rv1(16);
+    result1.visit([&](auto output) { rv1.assign(output.begin(), output.end()); });
+    auto result2 = p2.eval({{"input", input}, {"weights", weights}}).back();
+    std::vector<float> rv2(16);
+    result2.visit([&](auto output) { rv2.assign(output.begin(), output.end()); });
+    EXPECT(migraphx::verify_range(rv1, rv2));
+}
+
+TEST_CASE(dot_correctness)
+{
+    migraphx::shape sh1{migraphx::shape::float_type, {10, 4}};
+    migraphx::shape sh2{migraphx::shape::float_type, {4, 12}};
+    migraphx::shape sh3{migraphx::shape::float_type, {10, 12}};
+
+    migraphx::program p1;
+    {
+        auto* m1     = p1.get_main_module();
+        auto a       = m1->add_parameter("a", sh1);
+        auto b       = m1->add_parameter("b", sh2);
+        auto c       = m1->add_parameter("c", sh3);
+        auto scale_a = m1->add_literal(0.4f);
+        auto scale_b = m1->add_literal(0.5f);
+        auto zero    = m1->add_literal(std::int8_t{0});
+
+        auto q1 = add_quantize_op(*m1, "quantizelinear", a, scale_a, zero);
+        auto d1 = add_quantize_op(*m1, "dequantizelinear", q1, scale_a, zero);
+        auto q2 = add_quantize_op(*m1, "quantizelinear", b, scale_b, zero);
+        auto d2 = add_quantize_op(*m1, "dequantizelinear", q2, scale_b, zero);
+        auto dot =
+            m1->add_instruction(migraphx::make_op("dot", {{"alpha", 1}, {"beta", 0}}), d1, d2, c);
+        m1->add_return({dot});
+
+        run_pass(*m1);
+    }
+
+    migraphx::program p2;
+    {
+        auto* m2 = p2.get_main_module();
+        auto a   = m2->add_parameter("a", sh1);
+        auto b   = m2->add_parameter("b", sh2);
+        auto c   = m2->add_parameter("c", sh3);
```

Comment:
since beta is 0, so we do not need 'c' anymore.

Reply:
In the file parse_gemm.cpp, there is one change needed to set beta always to be 0. The last few lines should be changed to:

```
        l1      = (transa) ? info.add_instruction(make_op("transpose", {{"dims", perm}}), l1) : l1;
        auto l2 = (transb) ? info.add_instruction(make_op("transpose", {{"dims", perm}}), args[1])
                           : args[1];

        auto ret = info.add_instruction(make_op("dot", {{"alpha", 1.0f}, {"beta", 0.0f}}), l1, l2);

        if(args.size() == 3)
        {
            if(not float_equal(beta, 0.0f) && args[2]->get_shape().elements() > 0)
            {
                auto out_lens   = l1->get_shape().lens();
                out_lens.back() = l2->get_shape().lens().back();
                auto l3         = args[2];
                auto l3_lens    = l3->get_shape().lens();
                if(!std::equal(out_lens.begin(), out_lens.end(), l3_lens.begin(), l3_lens.end()))
                {
                    l3 = info.add_instruction(
                        make_op("multibroadcast", {{"output_lens", out_lens}}), args[2]);
                }
                auto beta_literal = info.add_literal(beta);
                auto beta_l3      = info.add_broadcastable_binary_op("mul", l3, beta_literal);
                if(beta_l3->get_shape().type() != dot_type)
                {
                    beta_l3 = info.add_instruction(make_op("convert", {{"target_type", dot_type}}),
                                                   beta_l3);
                }

                ret = info.add_instruction(make_op("add"), ret, beta_l3);
            }
        }
```

Of course, the related to gemm onnx_test should be changed accordingly.